File: java/src/Personne.java
public class Personne {
    private String nom;
    private int age;

    public Personne(String nom, int age) {
        setNom(nom);
        setAge(age);
    }

    public String getNom() { return nom; }
    public int getAge() { return age; }

    public void setNom(String nom) {
        if (nom == null || nom.trim().isEmpty())
            throw new IllegalArgumentException("Le nom ne peut pas être vide");
        this.nom = nom;
    }

    public void setAge(int age) {
        if (age < 0 || age > 100)
            throw new IllegalArgumentException("L'âge doit être entre 0 et 100");
        this.age = age;
    }

    public void afficherDetails() {
        System.out.println(this);
    }

    @Override
    public String toString() {
        return "Personne: " + nom + ", " + age + " ans";
    }
}

File: Module-1-TP-HEP/python/src/personne.py
from abc import abstractmethod
from .interfaces import Affichable

class Personne(Affichable):
    def __init__(self, nom: str, age: int):
        self.nom = nom
        self.age = age

    @property
    def nom(self) -> str:
        return self._nom

    @nom.setter
    def nom(self, valeur: str):
        if not valeur or not valeur.strip():
            raise ValueError("Le nom ne peut pas être vide")
        self._nom = valeur

    @property
    def age(self) -> int:
        return self._age

    @age.setter
    def age(self, valeur: int):
        if valeur < 0 or valeur > 100:
            raise ValueError("L'âge doit être entre 0 et 100")
        self._age = valeur

    # Chaque sous-classe DOIT implémenter afficher_details() (OCP + ISP)
    @abstractmethod
    def afficher_details(self) -> None:
        pass

    def __str__(self):
        return f"Personne: {self.nom}, {self.age} ans"
